specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-08-01/frontdoor.json

```diff
@@ -0,0 +1,2484 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-08-01",
+    "title": "FrontDoorManagementClient",
+    "description": "Use these APIs to manage Azure Front Door resources through the Azure Resource Manager. You must make sure that requests made to these resources are secure."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.Network/checkFrontDoorNameAvailability": {
```

Comment:
As per ARM RPC, checkNameAvailability should be modeled as pointed out below - https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#check-name-availability-requests